axelor-account/src/main/java/com/axelor/apps/account/service/move/template/MoveTemplateGroupServiceImpl.java

```diff
@@ -0,0 +1,107 @@
+package com.axelor.apps.account.service.move.template;
+
+import com.axelor.apps.account.db.Journal;
+import com.axelor.apps.account.db.MoveTemplate;
+import com.axelor.apps.account.db.MoveTemplateType;
+import com.axelor.apps.account.db.repo.JournalRepository;
+import com.axelor.apps.account.db.repo.MoveTemplateRepository;
+import com.axelor.apps.account.db.repo.MoveTemplateTypeRepository;
+import com.axelor.apps.base.AxelorException;
+import com.axelor.apps.base.db.Company;
+import com.axelor.apps.base.db.repo.CompanyRepository;
+import com.axelor.apps.base.service.app.AppBaseService;
+import com.axelor.apps.base.service.user.UserService;
+import com.google.inject.Inject;
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+
+public class MoveTemplateGroupServiceImpl implements MoveTemplateGroupService {
+
+  protected CompanyRepository companyRepository;
+  protected UserService userService;
+  protected MoveTemplateTypeRepository moveTemplateTypeRepository;
+  protected MoveTemplateRepository moveTemplateRepository;
+  protected AppBaseService appBaseService;
+
+  @Inject
+  public MoveTemplateGroupServiceImpl(
+      CompanyRepository companyRepository,
+      UserService userService,
+      MoveTemplateTypeRepository moveTemplateTypeRepository,
+      MoveTemplateRepository moveTemplateRepository,
+      AppBaseService appBaseService) {
+    this.companyRepository = companyRepository;
+    this.userService = userService;
+    this.moveTemplateTypeRepository = moveTemplateTypeRepository;
+    this.moveTemplateRepository = moveTemplateRepository;
+    this.appBaseService = appBaseService;
+  }
+
+  @Override
+  public Map<String, Object> getOnNewValuesMap(
+      Long companyId, Long moveTemplateTypeId, Long moveTemplateOriginId) throws AxelorException {
+    Map<String, Object> valuesMap = new HashMap<>();
+
+    MoveTemplateType moveTemplateType = moveTemplateTypeRepository.find(moveTemplateTypeId);
+    Company company = getDefaultCompany(companyId);
+    MoveTemplate moveTemplate = getDefaultMoveTemplate(moveTemplateOriginId, company);
+
+    valuesMap.put("company", company);
+
+    if (moveTemplate != null
+        && moveTemplateType != null
+        && moveTemplateType.getTypeSelect() != null) {
+      valuesMap.put("moveTemplateType", moveTemplate.getMoveTemplateType());
+      if (moveTemplateType.getTypeSelect() == MoveTemplateTypeRepository.TYPE_PERCENTAGE) {
+        valuesMap.put("moveTemplate", moveTemplate);
+      } else {
+        valuesMap.put("moveDate", appBaseService.getTodayDate(company));
+        valuesMap.put("moveTemplateSet", new HashSet<>(Arrays.asList(moveTemplate)));
+      }
+      valuesMap.put("popup", true);
```

Comment:
I don't know if this is a mistake. If this parameter is useful, then it should be on the controller as it is strictly related to the view.